File: code/accounts/program-derived-address/derived-a-pda/find-program-address.en.ts
import { PublicKey } from "@solana/web3.js";

(async () => {
  const programId = new PublicKey("G1DCNUQTSGHehwdLCAmRyAG8hf51eCHrLNUqkgGKYASj");

  let [pda, bump] = await PublicKey.findProgramAddress([Buffer.from("test")], programId);
  console.log(`bump: ${bump}, pubkey: ${pda.toBase58()}`);
  // you will find the result is different from `createProgramAddress`.
  // It is expected because the real seed we used to calculate is ["test" + bump]
})();

File: code/accounts/program-derived-address/derived-a-pda/find-program-address.en.rs
use solana_program::pubkey::Pubkey;
use std::str::FromStr;

fn main() {
    let program_id = Pubkey::from_str("G1DCNUQTSGHehwdLCAmRyAG8hf51eCHrLNUqkgGKYASj").unwrap();

    let (pda, bump_seed) = Pubkey::find_program_address(&[b"test"], &program_id);
    println!("pda: {}, bump: {}", pda, bump_seed);
}
